src/Host/Broker/Impl/ApiErrorResult.cs

```diff
@@ -0,0 +1,15 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.R.Host.Protocol;
+
+namespace Microsoft.R.Host.Broker {
+    internal sealed class ApiErrorResult : ObjectResult {
+        public ApiErrorResult(HttpResponse response, BrokerApiError error) : base(error) {
+            StatusCode = StatusCodes.Status500InternalServerError;
```

Comment:
Yeah, ASP.NET uses 500 if you let an exception escape the controller. It's because we're technically not supposed to do that ever, so it's considered an internal server bug :)

412 is actually for a very specific kind of precondition failure - it's only when the client specified a precondition HTTP header (`If-Match`), and the server couldn't honor it. It's not supposed to be used for generic kind of precondition errors.

Note that the RFC 2616 definition of 400 ("malformed syntax") is outdated. The current definitions are in this one:
https://tools.ietf.org/html/rfc7231#section-6.5.1

> The 400 (Bad Request) status code indicates that the server cannot or will not process the request due to _something that is perceived to be a client error_ (e.g., malformed request syntax, invalid request message framing, or deceptive request routing).

Hence why it was adopted as a generic error code for this sort of stuff in REST. I think the only others that are used with any regularity are 403 and 404, but they only apply in very narrow circumstances.